src/main/java/org/folio/service/rollover/LedgerRolloverService.java

```diff
@@ -105,15 +107,16 @@ public Future<Void> rolloverPreparation(LedgerFiscalYearRollover rollover, Ledge
       RequestContext requestContext) {
     DBClient client = requestContext.toDBClient();
     return client.startTx()
-      .compose(aVoid -> ledgerFiscalYearRolloverDAO.create(rollover, client))
-      .compose(aVoid -> rolloverProgressService.createRolloverProgress(progress, client))
-      .compose(aVoid -> fiscalYearService.populateRolloverWithCurrencyFactor(rollover, requestContext))
-      .compose(aVoid -> closeBudgets(rollover, client))
-      .compose(aVoid -> client.endTx())
-      .onFailure(t -> {
-        log.error("Rollover preparation failed for Ledger {}", rollover.getLedgerId(), t);
-        client.rollbackTransaction();
-      });
+      .compose(aVoid -> uniqueValidationService.validationOfUniqueness(rollover, client))
+      .compose(v -> ledgerFiscalYearRolloverDAO.create(rollover, client)
+        .compose(aVoid -> rolloverProgressService.createRolloverProgress(progress, client))
```

Comment:
indentation seems a little off compared to the previous line.

Reply:
This is because it is nested compose method.